wolfcrypt/test/test.c

```diff
@@ -13556,9 +13560,50 @@ static wc_test_ret_t simple_mem_test(int sz)
 }
 #endif
 
+/* If successful, returns the first letter of the byte array `in`.
+**
+** This is a deceptively simple test of a read-only embedded Linux file system.
+** (e.g CFLAGS `-mfdpic` and `-mforce-l32`) When `-mforce-l32` is missing,
+** access to `in` will fail as the const is on the R/O memory-mapped file
+** system. Const is *not* loaded into app memory. See PR 6523.
+**
+** Edit with caution. */
+static wc_test_ret_t const_byte_ptr_test(const byte* in, word32 *outJ)
+{
+    volatile wc_test_ret_t ret;
```

Comment:
Do you really understand the use of `volatile` here?

Reply:
I think so. I was having problems with the compiler optimizing away things.

Reply:
I was a bit too gratuitous on the `volatile`; I've removed the one from `wc_test_ret_t ret` but the next one is essential on `volatile word32 j`; without the `volatile` keyword, I've confirmed the `Illegal instruction` error does not occur.